specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/searchservice.json

```diff
@@ -4383,6 +4383,13 @@
             "url": "https://docs.microsoft.com/rest/api/searchservice/Custom-analyzers-in-Azure-Search"
           }
         },
+        "encryptionKey": {
+          "$ref": "#/definitions/EncryptionKey",
+          "description": "The configuration for encryption with customer managed keys. On updates, the encryptionKey will stay unchanged if not specified. Encryption with customer managed keys is only available on paid services created after 01/01/2019.",
```

Comment:
Should it be "customer-managed keys" (with a dash)? @HeidiSteen what do you think?

Also:
  - Maybe change the first sentence to make it more obvious when a user would need to care about this. How about: "The configuration to use when you want to encrypt the index with your own encryption key."
  - Re: "On updates, the encryptionKey…": Mentioning property names in the descriptions is dicey since the names are different depending on the target language. I prefer to keep things in terms of the concepts, or just not use names. In this case, maybe just say "On updates, this property will stay unchanged..." Also, in terms of the wire format, what happens when the property is explicitly set to null?
  - "paid services" is easy to miss, and doesn't reflect terminology we use with customers. It's probably better to call it out explicitly in contrast to free services. How about: "Encryption with customer-managed keys is not available for free services. For paid services, it is only available for search services created after 01/01/2019".
  - Regarding the date, is it really after Jan 1st, or "on or after" Jan 1st? Also, maybe format the date in ISO format to make it regionally unambiguous, like 2019-01-01.

Reply:
how about this -> The configuration to use when you want to encrypt the index with your own encryption key (customer-managed key). On updates, the encryption key configuration will stay unchanged if not specified. Encryption with customer-managed keys is not available for free services. For paid services, it is only available for services created on or after 2019-01-1.

About the null value, as far as I know, and correct me if I'm wrong, we make no distinction between properties that were omitted in the json payload or property that were explicitely set to null once we deserialize them in our controller (I think this applies to all our public models). For encryption keys, unspecified or null property both means "don't change anything". I didn't explicitely call out "null" in the description as it seemed too language specific.

Reply:
@shmed The updated text looks good. Regarding null, I think it's fine to call it out since JSON has null and every language will have its equivalent concept.